### PR TITLE
7919: Create EligibleCase entity to whitelist eligible cases data

### PR DESCRIPTION
**Pull request by sutt0n**
This new entity proposes a slimmer entity to address the bug ustaxcourt/ef-cms#7744 in which a large payload is being returned on the `eligible-cases` endpoint.

### CODE REVIEW

**Review by kkoskelin — CHANGES_REQUESTED** [review]
Everything looks good so far. Once the interactor tests are up, I'll thumb it. 👍

File: shared/src/test/mockCase.js [hunk 1]
```diff
@@ -211,3 +211,65 @@ exports.MOCK_CASE_WITH_TRIAL_SESSION = {
...
+        address2: 'Apt. 835',
+        address3: null,
+        city: 'Cristianville',
+        country: 'U.S.A.',
+        countryType: 'domestic',
+        phone: '001-016-669-6532x5946',
+        postalCode: '68117',
+        state: 'NE',
+      },
+      email: 'adam22@yahoo.com',
```
Comment (rachaelparris) [inline shared/src/test/mockCase.js]: `example.com` please 😄

File: shared/src/test/mockCase.js [hunk 2]
```diff
@@ -211,3 +211,65 @@ exports.MOCK_CASE_WITH_TRIAL_SESSION = {
...
+        address2: 'Suite 334',
+        address3: null,
+        city: 'Millermouth',
+        country: 'U.S.A.',
+        countryType: 'domestic',
+        phone: '348-858-8312',
+        postalCode: '99517',
+        state: 'AK',
+      },
+      email: 'thomastorres@yahoo.com',
```
Comment (rachaelparris) [inline shared/src/test/mockCase.js]: Also `example.com`

**Review by kkoskelin — APPROVED** [review]
Address comments and we're good to go. 👍

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**